lib/util.ts

```diff
@@ -0,0 +1,168 @@
+import axios from 'axios';
+import { newMessage } from './messages';
+import { AxiosRequestHeaders } from 'axios';
+import { transform } from '@openintegrationhub/ferryman';
+import { Config, GenericObject, Message, Self, Auth, Headers } from './types/global';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export async function processMethod(self: Self, msg: Message, cfg: Config, snapshot?: GenericObject) {
+    const { endpointUrl, soapAction, httpHeaders, soapHeaders } = cfg;
+    const xml = msg.data.xml;
+
+    const { auth } = getAuthFromSecretConfig(cfg, self);
+    const bearerToken = (auth && auth.oauth2 && auth.oauth2.keys && auth.oauth2.keys.access_token ? auth.oauth2.keys.access_token : '');
+
+    let requestHeaders: Array<Headers> = [];
+    if (auth) {
+      requestHeaders = populateAuthHeaders(auth, self, bearerToken, httpHeaders);
+    }
+    requestHeaders.push(
+      {
+          key: 'content-type',
+          value: 'text/xml;charset=UTF-8'
+      }
+    )
+
+    if (soapAction) {
```

Comment:
Some of this logic following could be delegated to a soap module to create the request.